src/cassandra/future.rs

```diff
@@ -123,6 +127,54 @@ impl Completable for PreparedStatement {
     }
 }
 
+/// For each custom payload defined in the future, convert it into a
+/// name and value pair, then insert it into the CustomPayloadResponse.
+///
+unsafe fn payloads_from_future(future: *mut _Future) -> Result<CustomPayloadResponse> {
+    let cp_count = cass_future_custom_payload_item_count(future);
+    (0..cp_count)
+        .into_iter()
+        .map(|index| {
+            let mut name = mem::zeroed();
+            let mut name_length = mem::zeroed();
+            let mut value = mem::zeroed();
+            let mut value_size = mem::zeroed();
+
+            cass_future_custom_payload_item(
+                future,
+                index,
+                &mut name,
+                &mut name_length,
+                &mut value,
+                &mut value_size,
+            )
+            .to_result((name, name_length, value, value_size))
+            .and_then(|(name, name_length, value, value_size)| {
+                let name_slice = slice::from_raw_parts(name as *const u8, name_length);
+                str::from_utf8(name_slice)
+                    .map_err(|err| err.into())
+                    .map(|name| {
+                        (
+                            name.to_string(),
+                            slice::from_raw_parts(value, value_size).to_vec(),
+                        )
```

Comment:
Nice. I confirm I've reviewed the C++ docs and this code and it looks to me like you're doing all the right things.

Reply:
👍